common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -405,19 +469,35 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
+                    result = DeadlineRunResult.TIMED_OUT;
```

Comment:
I don't quiet agree with this approach being strictly _better_. Yes it removes the enum of return type but introduces a complexity that ties a callback of execution time and return value. 
I consider this callback as a side-effect that happens to impact `SingleThreadEventLoop` but not when called from `SingleThreadEventExecutor`.

IOW, I think these two are different approaches and one is not strictly better than other.